lib/utils/aws.js

```diff
@@ -84,15 +84,13 @@ module.exports.profilesMap = function() {
  * @param secretKey
  */
 module.exports.profilesSet = function(awsProfile, awsRegion, accessKeyId, secretKey) {
-  var credsPath = path.join(this.getConfigDir(), 'credentials'),
-      configPath = path.join(path.dirname(credsPath), 'config');
+  utils.jawsDebug('Setting new AWS profile:', awsProfile);
+  var configDir = this.getConfigDir(),
+      credsPath = path.join(configDir, 'credentials'),
+      configPath = path.join(configDir, 'config');
 
-  if (!utils.fileExistsSync(credsPath)) {
-    utils.writeFile(credsPath);
-  }
-
-  if (!utils.fileExistsSync(configPath)) {
-    utils.writeFile(configPath);
+  if (!utils.dirExistsSync(configDir)) {
+    fs.mkdirSync(configDir, 0o700);
```

Comment:
Whoops sorry about that. Using node 4.1 locally so I didn't even realize.

Thanks for fixing, just the number format?

Aaron

On Fri, Sep 25, 2015, 2:35 PM Ryan Pendergast <notifications@github.com>
wrote:

> In lib/utils/aws.js
> <https://github.com/jaws-framework/JAWS/pull/153#discussion_r40477980>:
>
> >
> > -  if (!utils.fileExistsSync(credsPath)) {
> > -    utils.writeFile(credsPath);
> > -  }
> > -
> > -  if (!utils.fileExistsSync(configPath)) {
> > -    utils.writeFile(configPath);
> > +  if (!utils.dirExistsSync(configDir)) {
> > +    fs.mkdirSync(configDir, 0o700);
>
> Just a heads up, we are not forcing es6 yet. had to change this to es5
> compat.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jaws-framework/JAWS/pull/153/files#r40477980>.
>